src/Our.Umbraco.Ditto/ComponentModel/TypeConverters/HtmlStringConverter.cs

```diff
@@ -0,0 +1,81 @@
+﻿namespace Our.Umbraco.Ditto
+{
+    using System;
+    using System.ComponentModel;
+    using System.Globalization;
+    using System.Web;
+
+    using global::Umbraco.Web;
+    using global::Umbraco.Web.Templates;
+
+    /// <summary>
+    /// Provides a unified way of converting <see cref="String"/>s to <see cref="HtmlString"/>'s.
+    /// </summary>
+    public class HtmlStringConverter : TypeConverter
+    {
+        /// <summary>
+        /// Returns whether this converter can convert an object of the given type to the type of this converter,
+        /// using the specified context.
+        /// </summary>
+        /// <param name="context">
+        /// An <see cref="T:System.ComponentModel.ITypeDescriptorContext" /> that provides a format context.
+        /// </param>
+        /// <param name="sourceType">
+        /// A <see cref="T:System.Type" /> that represents the type you want to convert from.
+        /// </param>
+        /// <returns>
+        /// true if this converter can perform the conversion; otherwise, false.
+        /// </returns>
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+            if (sourceType == typeof(string) || sourceType == typeof(HtmlString))
```

Comment:
I can extend this one, as Umbraco core can return rich-text editor value as `Umbraco.Core.Dynamics.DynamicXml`.
(All part of the mystical... [`ConvertUsingDarkMagic`](https://github.com/umbraco/Umbraco-CMS/blob/dev-v7/src/Umbraco.Core/Models/PublishedContent/PublishedPropertyType.cs#L324) :smiling_imp:)